openid4vc-high-assurance-interoperability-profile-sd-jwt-vc-1_0.md

```diff
@@ -215,6 +223,57 @@ This is an example of a Wallet Instance Attestation:
     * In the `input-descriptors` object, `id`, `name`, `purpose`, `group`, `format`, and `constraints` properties MUST be supported. In the `constraints` object, `limit_disclosure`, and `fields` properties MUST be supported. In the `fields` object, `path` and `filter` properties MUST be supported. A `path` MUST contain exactly one entry with a static path to a certain claim. A `filter` MUST only contain `type` elements of value `string` and `const` elements.
     * In the `submission_requirements` object, `name`, `rule (`pick` only)`, `count`, `from` properties MUST be supported.
 
+# OpenID for Verifiable Presentations over W3C Digital Credentials API
+
+* MUST support Annex A in [@!OIDF.OID4VP] that defines how to use OID4VP over the W3C Digital Credentials API.
+  * Annex A.3.1. in [@!OIDF.OID4VP] MUST NOT be used. OID4VP requests over the W3C Digital Credentials API MUST be signed.
+* Wallet Invocation is done via the W3C Digital Credentials API or an equivalent platform API. Custom URL schemes MUST NOT be used.
+* Response Mode MUST be `w3c_dc_api.jwt`. Encryption of the response is mandatory.
+* MUST support Transaction Data defined in Sections 5.4 and 7.4 of [@!OIDF.OID4VP].
+* The DQCL query and response as defined in Section 6 of [@!OIDF.OID4VP] MUST be used. Presentation Exchange as defined in Sections 5.4 and 5.5 of [@!OIDF.OID4VP] MUST NOT be used. Below is the list of features in the DQCL query and response that MUST be supported:
+  * tbd
+
+## ISO mdoc specific requirements for OpenID for Verifiable Presentations over W3C Digital Credentials API
+
+  * The Credential format identifier MUST be `mso_mdoc`.
+  * mdoc Credential Format specific DCQL parameters as defined in Section 6.4.2. of [@!OIDF.OID4VP] MUST be used.
+  * Verifier MAY request more than one credential in the same request.
+  * DeviceResponse in the VP Token MUST contain only one mdoc. Therefore, when multiple mdocs are being returned, each mdoc should be returned in a separate DeviceResponse, each matching to a respective DCQL query.
```

Comment:
not sure I follow. but yes, there might be multiple deviceResponse objects in a VP Token, but given the way DCQL and VP Token is define, there should be only one mdoc per deviceResponse.

Reply:
Note that there is an issue that is slightly related to this: https://github.com/openid/OpenID4VP/issues/336
That issue is the other way round, but it may result in changing the structure / requirements for how to respond to a query.

Reply:
that's what I meant

Reply:
that is what this text intends to say. are there ways to make that clearer?

```suggestion
  * Each DeviceResponse in the VP Token MUST contain one mdoc. Therefore, when multiple mdocs are being returned, each mdoc should be returned in a separate DeviceResponse, each matching to a respective DCQL query.
```